src/main/scala/temple/test/ProjectTester.scala

```diff
@@ -17,27 +19,56 @@ object ProjectTester {
 
   /** Configure the infrastructure for testing */
   private def performSetup(templefile: Templefile, generatedPath: String): ProjectConfig = {
-    println("🍪 Spinning up Kubernetes infrastructure...")
-    // Deploy Kubernetes
-    exec(s"sh $generatedPath/deploy.sh")
+    val provider = templefile
+      .lookupMetadata[Metadata.Provider]
+      .getOrElse(throw new RuntimeException("Could not find deployment information"))
 
-    // Grab Kong's URL for future requests
-    val Array(baseURL, kongAdmin, _*) = exec("minikube service kong --url").split("\n")
-    val baseIP                        = baseURL.replaceAll("http[s]*://", "")
-    val config                        = ProjectConfig(baseIP, kongAdmin)
+    val config = provider match {
+      case Provider.Kubernetes =>
+        println("🍪 Spinning up Kubernetes infrastructure...")
+        exec(s"sh $generatedPath/deploy.sh")
+
+        // Grab Kong's URL for future requests
+        val Array(baseURL, kongAdmin, _*) = exec("minikube service kong --url").split("\n")
+        val baseIP                        = baseURL.replaceAll("http[s]*://", "")
+        ProjectConfig(baseIP, kongAdmin)
+      case Provider.DockerCompose =>
+        println("🐳 Spinning up Docker Compose infrastructure...")
+        exec(
+          s"cd $generatedPath && docker ps -a -q | xargs docker rm -f && docker volume prune -f && docker-compose up --build -d",
+        )
+
+        var finishedStarting = false
+        while (!finishedStarting) {
+          val kongRequest = Try(Http("http://localhost:8001/status").asString)
```

Comment:
Is it safe to use Try? Could there be any other exceptions that we don’t want to discard?